codec-http2/src/main/java/io/netty/handler/codec/http2/HttpToHttp2ConnectionHandler.java

```diff
@@ -60,6 +70,9 @@ protected HttpToHttp2ConnectionHandler(Http2ConnectionDecoder decoder, Http2Conn
      * @throws Exception If the {@code httpHeaders} object specifies an invalid stream id
      */
     private int getStreamId(HttpHeaders httpHeaders) throws Exception {
+        if (httpScheme != null && !httpHeaders.contains(HttpConversionUtil.ExtensionHeaderNames.SCHEME.text())) {
+            httpHeaders.set(HttpConversionUtil.ExtensionHeaderNames.SCHEME.text(), httpScheme.name());
```

Comment:
honestly I think thats a very bad idea... a method with `get` in its name should not modify the headers that are passed in. This is just super unexpected.

Reply:
Agree. Should I move into the `write` method?

Reply:
yes

Reply:
Also please add a test case

Reply:
@hyperxpro please ping me once done

Reply:
Sorry for the delay. Been really busy. I made changes as you told and also added a test case.